python/ray/serve/config.py

```diff
@@ -258,15 +258,17 @@ def grpc_servicer_func_callable(self) -> List[Callable]:
                 if callable(imported_func):
                     callables.append(imported_func)
                 else:
-                    logger.warning(
+                    logger.exception(
                         f"{func} is not a callable function! Please make sure "
                         "the function is imported correctly."
                     )
-            except ModuleNotFoundError:
-                logger.warning(
+                    raise ValueError(f"{func} is not a callable function!")
+            except ModuleNotFoundError as e:
+                logger.exception(
```

Comment:
I would suggest wrapping this info in the exception message rather than logging it w/ `logger` here to make sure the exception message is fully self-contained. Also users might disable the logger or redirect it to a file in which case they may not find the detailed msg